client/data/themes/use-active-theme-query.ts

```diff
@@ -0,0 +1,28 @@
+import { useQuery, UseQueryResult } from 'react-query';
+import wpcom from 'calypso/lib/wp';
+
+interface ThemeSupports {
+	[ index: string ]: boolean;
+}
+
+export type ActiveTheme = {
+	theme_supports: ThemeSupports;
+};
+
+export const useActiveThemeQuery = (
+	siteId: number,
+	isEnabled = false
+): UseQueryResult< ActiveTheme[] > => {
+	const queryKey = [ 'activeTheme', siteId ];
```

Comment:
I wonder if we could/should include the theme slug here in the cache key, as well, so that the cache is busted when the theme is switched?

However, in practice this doesn't seem to be needed. If I open My Home in one tab, and change themes in another tab, the Edit site link instantly is displayed/hidden after theme switch--it appears there's already something busting the cache when the theme changes.